liesel/model/model.py

```diff
@@ -675,9 +673,13 @@ def __init__(
             nodes_and_vars = [*model.nodes.values(), *model.vars.values()]
             model.pop_nodes_and_vars()
 
+        groups = {g for nv in nodes_and_vars for g in nv.groups.values()}
         self._nodes = {nv.name: nv for nv in nodes_and_vars if isinstance(nv, Node)}
         self._vars = {nv.name: nv for nv in nodes_and_vars if isinstance(nv, Var)}
 
+        if len({g.name for g in groups}) < len(groups):
+            raise RuntimeError("Model received groups with duplicate names")
```

Comment:
Small thing: I think the error message is more helpful, if the duplicate name shows up in there.

Reply:
As you wish! :rocket: